File: 2403/2403.c
int minimumTimeToKillAllMonsters(struct TreeNode* root) {
    if (!root) return 0;
    int l = minimumTimeToKillAllMonsters(root->left);
    int r = minimumTimeToKillAllMonsters(root->right);
    return 1 + (l > r ? l : r);
}

File: 2403/2403.go
func minimumTimeToKillAllMonsters(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := minimumTimeToKillAllMonsters(root.Left)
	r := minimumTimeToKillAllMonsters(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}